### DISCUSSION TITLE
Broken on vite?

### DISCUSSION THREAD

**Discussion by An-Huynh-Certivity** [wojtekmaj/react-pdf#1376] [Q&A]
Turns out I am simply too stupid to read the docs, more specifically this part
![image](https://user-images.githubusercontent.com/100586735/224163672-c9c71b3c-7f3c-4c97-96e8-032af1c8b975.png)

**Comment by AlexMachin1997** [comment]
Your not too stupid 😉 You should just be able to import the library e.g. "react-pdf" and it should work none of this weird entry file nonsense.

What makes this worse is the 1mb it adds to the bundle of the project.......

**Reply by wojtekmaj** [reply]
I agree it's nonsense! The sad truth is that in the past (which still haunts us as e.g. Create-React-App), there was no single way of importing a Worker from node_modules that worked on all browsers. They are much more aligned nowadays (although not perfect), so for example Webpack 5 and Vite entries are the identical, but Parcel? Webpack 4? Nope, we need something completely different. However, we still need to provide a generic entry that enables you to configure PDF.js by yourself e.g. by having worker served by a CDN, which (kinda) makes the bundle size problem slightly less of a deal. Slightly, because rendering PDFs is an enormous task, so it can't be done with a 20 kB dependency.

**Reply by AlexMachin1997** [reply]
Yeah I get it be a very difficult to build a library let alone which is complex like this one.

I think my biggest complaint is I can't easily take a binary format and just inject it, besides that it's fine. I want to handling data fetching myself via react-query but it's nothing but frustration atm.

**Comment by wojtekmaj** [comment]
Is there anything specific you don't understand and needs a hand with?
